pkg/controller/aws/identity/iamrole.go

```diff
@@ -0,0 +1,171 @@
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	awsiam "github.com/aws/aws-sdk-go-v2/service/iam"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	v1alpha1 "github.com/crossplaneio/crossplane/aws/apis/identity/v1alpha1"
+	"github.com/crossplaneio/crossplane/pkg/clients/aws/iam"
+	"github.com/crossplaneio/crossplane/pkg/controller/aws/utils"
+)
+
+const (
+	errObjectNotIAMRole = "The managed resource is not an IAMRole resource"
+)
+
+// IAMRoleController is the controller for IAMRole objects
+type IAMRoleController struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *IAMRoleController) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1alpha1.IAMRoleGroupVersionKind),
+		resource.WithExternalConnecter(&roleConnector{client: mgr.GetClient()}),
+		resource.WithManagedConnectionPublishers())
+
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1alpha1.IAMRoleKindAPIVersion, v1alpha1.Group))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.IAMRole{}).
+		Complete(r)
+}
+
+type roleConnector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) (iam.RoleClient, error)
+}
+
+func (conn *roleConnector) Connect(ctx context.Context, mgd resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRole)
+	if !ok {
+		return nil, errors.New(errObjectNotIAMRole)
+	}
+
+	awsconfig, err := utils.RetrieveAwsConfigFromProvider(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	newClientFn := iam.NewRoleClient
+	if conn.newClientFn != nil {
+		newClientFn = conn.newClientFn
+	}
+
+	c, err := newClientFn(awsconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create new IAMRole client")
+	}
+	return &roleExternal{c}, nil
+}
+
+type roleExternal struct {
+	client iam.RoleClient
+}
+
+func (e *roleExternal) Observe(ctx context.Context, mgd resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRole)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errObjectNotIAMRole)
+	}
+
+	req := e.client.GetRoleRequest(&awsiam.GetRoleInput{
+		RoleName: aws.String(cr.Spec.RoleName),
+	})
+	req.SetContext(ctx)
+
+	observed, err := req.Send()
+
+	if iam.IsErrorNotFound(err) {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	if err != nil {
+		return resource.ExternalObservation{}, err
+	}
+
+	cr.Status = cr.BuildStatusFromObservation(observed)
+
+	return resource.ExternalObservation{
+		ResourceExists: true,
+	}, nil
+}
+
+func (e *roleExternal) Create(ctx context.Context, mgd resource.Managed) (resource.ExternalCreation, error) {
```

Comment:
It just occurred to me that all create methods should be setting the below condition to reflect that the external resource is being created. Ditto for all the delete methods, except they should be setting `runtimev1alpha1.Deleting()`.

```go
mgd.SetConditions(runtimev1alpha1.Creating())
```